src/internal/puck/constant-value-fields/TextList.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+import { AutoField, IconButton, Button, CustomField } from "@measured/puck";
+import { Trash2 as TrashIcon } from "lucide-react";
+import { Plus as PlusIcon } from "lucide-react";
+
+export const TEXT_LIST_CONSTANT_CONFIG: CustomField<string[]> = {
+  type: "custom",
+  render: ({ onChange, value }) => {
+    const [localItems, setLocalItems] = useState<string[]>(value);
+
+    const updateItem = (index: number, value: string) => {
+      const updatedItems = [...localItems];
```

Comment:
This does a shallow clone so if it's a list of objects then it wouldn't work (would shallow clone level one but use a reference to the object deeper, in which case structuredClone is the answer), but these are strings so we should be ok.

Reply:
Yeah it should be fine for now. If we use this for other list types in the future we can update it then.